app/src/main/java/org/mozilla/focus/fragment/HomeFragment.kt

```diff
@@ -104,18 +106,23 @@ class HomeFragment : Fragment() {
     }
 }
 
-private class HomeTileAdapter(val onUrlEnteredListener: OnUrlEnteredListener, homeTiles: MutableList<HomeTile>) :
+private class HomeTileAdapter(val onUrlEnteredListener: OnUrlEnteredListener, val tiles: List<HomeTile>) :
         RecyclerView.Adapter<TileViewHolder>() {
-    val tiles = homeTiles
 
     override fun onBindViewHolder(holder: TileViewHolder, position: Int) = with (holder) {
         val item = tiles[position]
-        titleView.setText(item.title)
-        if (!item.imagePath.isNullOrBlank()) {
-            val bmImg = itemView.context.assets.open(DEFAULT_HOME_TILES_DIR + item.imagePath).use { BitmapFactory.decodeStream(it) }
-            iconView.setImageBitmap(bmImg)
+        when (item) {
+            is DefaultHomeTile -> {
```

Comment:
Why do you need the `is` again?

Reply:
`if (item is DefaultHomeTile)` – it's the operator to check the type of the variable. In Java, `is` would be `instanceof`